platform-metrics/src/test/java/org/hypertrace/core/serviceframework/metrics/PlatformMetricsRegistryTest.java

```diff
@@ -151,9 +152,40 @@ public void testDistributionSummary() {
     assertEquals(1, distribution.count());
     assertEquals(100, distribution.totalAmount());
     assertTrue(
-        Arrays.stream(distribution.takeSnapshot().percentileValues()).map(m -> m.percentile())
-            .collect(
-                Collectors.toList()).containsAll(List.of(0.5, 0.95, 0.99)));
+        Arrays.stream(distribution.takeSnapshot().percentileValues())
+            .map(m -> m.percentile())
+            .collect(Collectors.toList())
+            .containsAll(List.of(0.5, 0.95, 0.99)));
+  }
+
+  @Test
+  public void testCache() throws Exception {
+    initializeCustomRegistry(List.of("testing"));
+
+    Cache<String, Integer> cache = CacheBuilder.newBuilder().maximumSize(10).recordStats().build();
+    cache.put("One", 1);
+    Cache<String, Integer> monitor = PlatformMetricsRegistry.registerCache("my.cache", cache);
+    Callable<Integer> loader =
+        new Callable<Integer>() {
+          @Override
+          public Integer call() throws Exception {
+            return -1;
+          }
+        };
+    // Checking cache values
+    assertEquals(monitor.get("One", loader), 1);
+    cache.put("Two", 2);
+    assertEquals(monitor.get("Two", loader), 2);
+    assertEquals(cache.get("IsNotPresent", loader), -1);
+
+    // Checking cache stats
+    assertEquals(monitor.stats().hitCount(), 2);
```

Comment:
This seems more like a test on guava cache rather than the registry.

You should rather verify by getting the appropriate Meter objects from PlatformMetricsRegistry.getMeterRegistry().getMeters() and check if the cache hit/miss counts are captured correctly.

Reply:
I have updated the code, please review again.